infrastructures/docker/environment/src/main/java/org/eclipse/che/workspace/infrastructure/docker/environment/dockerimage/EntryPointParser.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.docker.environment.dockerimage;
+
+import static com.fasterxml.jackson.databind.node.JsonNodeType.STRING;
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.eclipse.che.api.core.model.workspace.config.MachineConfig;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.commons.annotation.Nullable;
+
+/**
+ * Can be used to parse container entry-point definition specified as a YAML list of strings or a
```

Comment:
See https://github.com/eclipse/che/pull/12761#discussion_r260754787. 
maybe `list of strings` would be enough here.

Reply:
You mean to not allow a simple string or have a custom format of "list of strings" not based on YAML?

Reply:
I mean allowing only list of strings, but not string type.
See fabric8 client kubernetes JSON Schema https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-model/kubernetes-model/src/main/resources/schema/kube-schema.json#L5198

I understand benefit of supporting YAML Strings (single string or list of strings) but we follow Kubernetes model here, and looks like they do not support it. For us, it would be much simpler too.

Reply:
:+1: